pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -235,6 +236,15 @@ func SetDefaults_Shoot(obj *Shoot) {
 	if obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication == nil {
 		obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication = pointer.BoolPtr(false)
 	}
+
+	if k8sVersionGreaterOrEqualThan122 {
+		for i := 0; i < len(obj.Spec.Provider.Workers); i++ {
```

Comment:
Any reason why it is not done in the usual golang way?
```suggestion
		for i, _ := range obj.Spec.Provider.Workers {
```

Reply:
This is what I originally had, then `gosimple` complained about the unnecessary `_` assignment at the end: https://github.com/gardener/gardener/pull/4222/commits/1f1494abdd2616eccb4c2ea5a2f4dcddb56d95a6

I guess the `gosimple` compatible solution using `range` would be 
```
for i := range obj.Spec.Provider.Workers {
```
and this felt kinda weird to me. I'm happy to change it, if you feel it is more idiomatic, though.

Reply:
Sorry, I haven't look in all commits to see that you have already tried.
Seems that `for i := range obj.Spec.Provider.Workers {` is exactly what you need to make the linter happy. The auto generated code is using the same approach https://github.com/gardener/gardener/blob/fe31ec9ff1f3d957cf2f848623384e7995720340/pkg/apis/core/v1beta1/zz_generated.conversion.go#L3870-L3872

Reply:
No worries, I can change it to use `range`. It felt weird when typing it, but that's probably just my lack of familiarity with Go code ;)